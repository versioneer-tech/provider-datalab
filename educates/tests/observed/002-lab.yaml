# Copyright 2025, EOX (https://eox.at) and Versioneer (https://versioneer.at)
# SPDX-License-Identifier: Apache-2.0

apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-secret-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: observe-secret-jeff
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: v1
      kind: Secret
      metadata:
        name: jeff
        namespace: datalab
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  watch: false
---
apiVersion: v1
kind: Secret
metadata:
  name: jeff
  namespace: datalab
  labels:
    crossplane.io/composite: s-jeff
type: Opaque
stringData:
  AWS_ACCESS_KEY_ID: jeff
  AWS_SECRET_ACCESS_KEY: testsecret123