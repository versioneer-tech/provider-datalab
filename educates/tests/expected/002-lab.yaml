---
apiVersion: pkg.internal/v1beta1
kind: Datalab
metadata:
  name: l002
  namespace: datalab
  uid: abc
spec:
  users:
  - u002a
  - u002b
  vcluster: "true"
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: keycloak-client-l002, keycloak-group-l002, keycloak-mapper-l002,
      and 6 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: openidclient.keycloak.m.crossplane.io/v1alpha1
kind: Client
metadata:
  annotations:
    crossplane.io/composition-resource-name: keycloak-client-l002
  labels:
    crossplane.io/claim-name: l002
    crossplane.io/composite: l002
  name: l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    accessType: PUBLIC
    adminUrl: https://l002.lab.acme.com
    clientId: l002
    directAccessGrantsEnabled: false
    implicitFlowEnabled: true
    name: l002
    oauth2DeviceAuthorizationGrantEnabled: true
    realmId: acme
    rootUrl: https://l002.lab.acme.com
    serviceAccountsEnabled: false
    standardFlowEnabled: true
    validRedirectUris:
    - https://l002.lab.acme.com/*
    - http://localhost:*
    webOrigins:
    - https://l002.lab.acme.com/*
    - http://localhost:*
  providerConfigRef:
    kind: ProviderConfig
    name: provider-keycloak
---
apiVersion: group.keycloak.m.crossplane.io/v1alpha1
kind: Group
metadata:
  annotations:
    crossplane.io/composition-resource-name: keycloak-group-l002
  labels:
    crossplane.io/composite: l002
  name: l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    name: l002
    realmId: acme
  providerConfigRef:
    kind: ProviderConfig
    name: provider-keycloak
---
apiVersion: openidgroup.keycloak.m.crossplane.io/v1alpha1
kind: GroupMembershipProtocolMapper
metadata:
  annotations:
    crossplane.io/composition-resource-name: keycloak-mapper-l002
  labels:
    crossplane.io/composite: l002
  name: l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    addToAccessToken: false
    addToIdToken: true
    addToUserinfo: false
    claimName: groups
    clientIdSelector:
      matchLabels:
        crossplane.io/claim-name: l002
    fullPath: false
    name: group-membership-mapper
    realmId: acme
  providerConfigRef:
    kind: ProviderConfig
    name: provider-keycloak
---
apiVersion: group.keycloak.m.crossplane.io/v1alpha1
kind: Memberships
metadata:
  annotations:
    crossplane.io/composition-resource-name: keycloak-memberships-l002
  labels:
    crossplane.io/composite: l002
  name: l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    groupIdRef:
      name: l002
      policy:
        resolution: Required
    members:
    - u002a
    - u002b
    realmId: acme
  providerConfigRef:
    kind: ProviderConfig
    name: provider-keycloak
---
apiVersion: role.keycloak.m.crossplane.io/v1alpha1
kind: Role
metadata:
  annotations:
    crossplane.io/composition-resource-name: keycloak-role-l002
  labels:
    crossplane.io/composite: l002
  name: l002-ws-access
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    clientIdRef:
      name: l002
      policy:
        resolution: Required
    name: ws_access
    realmId: acme
  providerConfigRef:
    kind: ProviderConfig
    name: provider-keycloak
---
apiVersion: group.keycloak.m.crossplane.io/v1alpha1
kind: Roles
metadata:
  annotations:
    crossplane.io/composition-resource-name: keycloak-roles-l002
  labels:
    crossplane.io/composite: l002
  name: l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    groupIdRef:
      name: l002
      policy:
        resolution: Required
    realmId: acme
    roleIdsRefs:
    - name: l002-ws-access
      policy:
        resolution: Required
  providerConfigRef:
    kind: ProviderConfig
    name: provider-keycloak
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: secret-observer-l002
  labels:
    crossplane.io/composite: l002
  name: secret-observer-l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    manifest:
      apiVersion: v1
      kind: Secret
      metadata:
        name: l002
        namespace: datalab
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  watch: true
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: workshop-l002
  labels:
    crossplane.io/composite: l002
  name: workshop-l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    manifest:
      apiVersion: training.educates.dev/v1beta1
      kind: Workshop
      metadata:
        name: l002
      spec:
        description: A default DataLab environment.
        environment:
          objects:
          - apiVersion: networking.k8s.io/v1
            kind: NetworkPolicy
            metadata:
              name: allow-dns
            spec:
              egress:
              - ports:
                - port: 53
                  protocol: UDP
                - port: 53
                  protocol: TCP
                to:
                - namespaceSelector:
                    matchLabels:
                      kubernetes.io/metadata.name: kube-system
                  podSelector: {}
              podSelector: {}
              policyTypes:
              - Egress
          - apiVersion: networking.k8s.io/v1
            kind: NetworkPolicy
            metadata:
              name: allow-session-to-data
            spec:
              ingress:
              - from:
                - namespaceSelector:
                    matchLabels:
                      training.educates.dev/component: session
                      training.educates.dev/environment.name: l002
                ports:
                - port: 8080
                  protocol: TCP
              podSelector:
                matchLabels:
                  app.kubernetes.io/instance: data
                  app.kubernetes.io/name: data
              policyTypes:
              - Ingress
          - apiVersion: v1
            kind: PersistentVolumeClaim
            metadata:
              name: l002-datalab
            spec:
              accessModes:
              - ReadOnlyMany
              resources:
                requests:
                  storage: 1Mi
              storageClassName: csi-rclone
          - apiVersion: apps/v1
            kind: Deployment
            metadata:
              labels:
                app.kubernetes.io/instance: data
                app.kubernetes.io/name: data
              name: data
            spec:
              replicas: 1
              selector:
                matchLabels:
                  app.kubernetes.io/instance: data
                  app.kubernetes.io/name: data
              template:
                metadata:
                  labels:
                    app.kubernetes.io/instance: data
                    app.kubernetes.io/name: data
                spec:
                  containers:
                  - env:
                    - name: FB_AUTH_MAPPER
                      value: writer-share
                    - name: WORKSPACE
                      valueFrom:
                        fieldRef:
                          fieldPath: metadata.namespace
                    - name: FB_ROOT
                      value: /data
                    - name: FB_BRANDING_NAME
                      valueFrom:
                        fieldRef:
                          fieldPath: metadata.namespace
                    - name: FB_AUTH_HEADER
                      value: Accept
                    - name: AWS_ACCESS_KEY_ID
                      valueFrom:
                        secretKeyRef:
                          key: AWS_ACCESS_KEY_ID
                          name: l002-datalab
                    - name: AWS_SECRET_ACCESS_KEY
                      valueFrom:
                        secretKeyRef:
                          key: AWS_SECRET_ACCESS_KEY
                          name: l002-datalab
                    - name: AWS_REGION
                      valueFrom:
                        secretKeyRef:
                          key: AWS_REGION
                          name: l002-datalab
                    - name: AWS_ENDPOINT_URL
                      valueFrom:
                        secretKeyRef:
                          key: AWS_ENDPOINT_URL
                          name: l002-datalab
                    - name: AWS_S3_FORCE_PATH_STYLE
                      valueFrom:
                        secretKeyRef:
                          key: AWS_S3_FORCE_PATH_STYLE
                          name: l002-datalab
                    image: ghcr.io/versioneer-tech/package-r:v2025.7.1
                    imagePullPolicy: Always
                    name: data
                    ports:
                    - containerPort: 8080
                      name: http
                    resources:
                      limits:
                        cpu: "1"
                        memory: 512Mi
                      requests:
                        cpu: 100m
                        memory: 128Mi
                    volumeMounts:
                    - mountPath: /data
                      name: data
                  serviceAccountName: $(service_account)
                  volumes:
                  - name: data
                    persistentVolumeClaim:
                      claimName: l002-datalab
          - apiVersion: v1
            kind: Service
            metadata:
              name: data
            spec:
              ports:
              - name: http
                port: 8080
                protocol: TCP
                targetPort: http
              selector:
                app.kubernetes.io/instance: data
                app.kubernetes.io/name: data
              type: ClusterIP
          secrets:
          - name: l002-datalab
            namespace: datalab
        session:
          applications:
            console:
              enabled: false
            editor:
              enabled: true
            slides:
              enabled: false
            terminal:
              enabled: true
            vcluster:
              enabled: true
            workshop:
              enabled: false
          dashboards:
          - name: Data
            url: $(ingress_protocol)://data-$(session_name).$(ingress_domain)/
          envFrom:
          - secretRef:
              name: l002-datalab
          ingresses:
          - host: data.$(workshop_namespace).svc.$(cluster_domain)
            name: data
            port: 8080
            protocol: http
          namespaces:
            budget: medium
            security:
              policy: baseline
          objects:
          - apiVersion: networking.k8s.io/v1
            kind: NetworkPolicy
            metadata:
              name: allow-dns
            spec:
              egress:
              - ports:
                - port: 53
                  protocol: UDP
                - port: 53
                  protocol: TCP
                to:
                - namespaceSelector:
                    matchLabels:
                      kubernetes.io/metadata.name: kube-system
                  podSelector: {}
              podSelector: {}
              policyTypes:
              - Egress
          - apiVersion: networking.k8s.io/v1
            kind: NetworkPolicy
            metadata:
              name: allow-to-environment-data
            spec:
              egress:
              - ports:
                - port: 8080
                  protocol: TCP
                to:
                - namespaceSelector:
                    matchLabels:
                      kubernetes.io/metadata.name: l002
                  podSelector:
                    matchLabels:
                      app.kubernetes.io/instance: data
                      app.kubernetes.io/name: data
              podSelector: {}
              policyTypes:
              - Egress
          resources:
            storage: 1Gi
        title: Default DataLab
        version: "1.0"
        workshop:
          files: []
          packages:
          - files:
            - image:
                url: ghcr.io/versioneer-tech/educates-extension-packages/awscli:v2.29.0
            name: awscli
          - files:
            - image:
                url: ghcr.io/versioneer-tech/educates-extension-packages/rclone:v1.71.0
            name: rclone
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: workshopenvironment-l002
  labels:
    crossplane.io/composite: l002
  name: workshopenvironment-l002
  namespace: datalab
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Datalab
    name: l002
    uid: abc
spec:
  forProvider:
    manifest:
      apiVersion: training.educates.dev/v1beta1
      kind: WorkshopEnvironment
      metadata:
        name: l002
      spec:
        analytics:
          amplitude:
            trackingId: ""
        cookies: {}
        environment:
          objects: []
          secrets: []
        request:
          enabled: false
        session:
          ingress:
            class: nginx
            domain: lab.acme.com
            secret: wildcard-tls
        theme:
          name: default-website-theme
        workshop:
          name: l002
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
  readiness:
    policy: SuccessfulCreate
  watch: false
